yt_dlp/extractor/bilibili.py

```diff
@@ -879,12 +899,26 @@ def _real_extract(self, url):
                 'Extracting episode', query={'fnval': 12240, 'ep_id': episode_id},
                 headers=headers))
 
+        geo_blocked = traverse_obj(play_info, (
+            'raw', 'data', 'plugins', lambda _, v: v['name'] == 'AreaLimitPanel', 'config', 'is_block', {bool}, any))
         premium_only = play_info.get('code') == -10403
-        play_info = traverse_obj(play_info, ('result', 'video_info', {dict})) or {}
 
-        formats = self.extract_formats(play_info)
-        if not formats and (premium_only or '成为大会员抢先看' in webpage or '开通大会员观看' in webpage):
-            self.raise_login_required('This video is for premium members only')
+        video_info = traverse_obj(play_info, (('result', ('raw', 'data')), 'video_info', {dict}, any)) or {}
+        formats = self.extract_formats(video_info)
+
+        if not formats:
+            if geo_blocked:
+                self.raise_geo_restricted()
+            elif premium_only or '成为大会员抢先看' in webpage or '开通大会员观看' in webpage:
+                self.raise_login_required('This video is for premium members only')
+
+        if traverse_obj(play_info, (
+            'result', 'play_check', 'play_detail', {lambda x: x and x == 'PLAY_PREVIEW'}),  # vs 'PLAY_WHOLE'
+            ('raw', 'data', 'play_video_type', {lambda x: x and x == 'preview'}),  # vs 'whole'
+        ):
```

Comment:
```suggestion
        if traverse_obj(play_info, ((
            ('result', 'play_check', 'play_detail'),
            ('raw', 'data', 'play_video_type'),
        ), any, {lambda x: x in ('PLAY_PREVIEW', 'preview')})):
```
perhaps something like this?

Reply:
we could retain the comments / info from the original code
```suggestion
        if traverse_obj(play_info, ((
            ('result', 'play_check', 'play_detail'),  # 'PLAY_PREVIEW' vs 'PLAY_WHOLE'
            ('raw', 'data', 'play_video_type'),  # 'preview' vs 'whole'
        ), any, {lambda x: x in ('PLAY_PREVIEW', 'preview')})):
```